hystrix/settings.go

```diff
@@ -101,3 +101,11 @@ func getSettings(name string) *settings {
 
 	return s
 }
+
+func GetCircuitSettingsMap() map[string]*settings{
+	copy := make(map[string]*settings)
+	for key, val := range circuitSettings {
```

Comment:
you should grab a read lock around the map before iterating.

`settingsMutex.RLock()`